src/Sql/dbo/Stored Procedures/GroupUser_UpdateUsers.sql

```diff
@@ -14,33 +14,56 @@ BEGIN
             [Id] = @GroupId
     )
 
-    ;WITH [AvailableUsersCTE] AS(
-        SELECT
-            [Id]
-        FROM
-            [dbo].[OrganizationUser]
-        WHERE
-            [OrganizationId] = @OrgId
+    CREATE TABLE #TempAvailableUsers
+    (
+        [Id] UNIQUEIDENTIFIER NOT NULL
     )
-    MERGE
-        [dbo].[GroupUser] AS [Target]
-    USING 
+
+    INSERT INTO #TempAvailableUsers
+    SELECT
+        [Id]
+    FROM
+        [dbo].[OrganizationUser]
+    WHERE
+        [OrganizationId] = @OrgId
+
+    -- Insert
+    INSERT INTO
+        [dbo].[GroupUser]
+    SELECT
+        @GroupId,
+        [Source].[Id]
+    FROM
         @OrganizationUserIds AS [Source]
-    ON
-        [Target].[GroupId] = @GroupId
-        AND [Target].[OrganizationUserId] = [Source].[Id]
-    WHEN NOT MATCHED BY TARGET
-    AND [Source].[Id] IN (SELECT [Id] FROM [AvailableUsersCTE]) THEN
-        INSERT VALUES
-        (
-            @GroupId,
-            [Source].[Id]
+    WHERE
+        [Source].[Id] IN (SELECT [Id] FROM #TempAvailableUsers)
+        AND NOT EXISTS (
+            SELECT
+                1
+            FROM
+                [dbo].[GroupUser]
+            WHERE
+                [GroupId] = @GroupId
+                AND [OrganizationUserId] = [Source].[Id]
+        )
+    
+    -- Delete
+    DELETE
+        GU
+    FROM
+        [dbo].[GroupUser] GU
+    WHERE
+        GU.[GroupId] = @GroupId
+        AND NOT EXISTS (
+            SELECT
```

Comment:
Using the `TOP 1` can sometimes increase performance acting as a `break` in a `SELECT` when you're only testing whether a value exists, e.g. `AND NOT EXISTS ( SELECT TOP 1 NULL FROM ... )`

Reply:
I think that this select is written in a way that will only ever resolve 1 result anyways.

Reply:
I can retract my statement as conjecture, just found this (it's old but likely still relevant): https://sqlinthewild.co.za/index.php/2011/04/05/to-top-or-not-to-top-an-exists/